samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -84,15 +87,17 @@
 
   protected static volatile boolean systemInitialized = false;
 
-  private Config config = null;
+  protected Config config = null;
   private SystemStream systemStream = null;
   private SystemProducer systemProducer = null;
   private String key = null;
-  private String streamName = null;
+  protected String streamName = null;
+  protected String appenderName = null;
+  private String containerName = null;
   private int partitionCount = 0;
   private boolean isApplicationMaster;
   private Serde<LogEvent> serde = null;
-  private Logger log = LogManager.getLogger(StreamAppender.class);
+  protected Logger log = LogManager.getLogger(StreamAppender.class);
```

Comment:
why does this need to be protected? can't subclasses have their own logger?

Reply:
See the comments thread in another rb.

Reply:
Sure we can keep the discussion there then circle back

Reply:
Get back from discussion in another rb: will change to each subclass has its own logger.